### ISSUE TITLE
Toolbar issue: Python3, wx4, windows only

### ISSUE THREAD

**Issue by briantoby** [matplotlib/matplotlib#9371]
### Bug report
**Bug summary**

Access to the toolbar fails on windows (but not Mac) with a wxPython backend using Phoenix (4.0.0b2) and matplotlib 2.0.2. (statement ```self.toolbar = Toolbar(self.canvas)``` throws an exception)

**Code for reproduction**

Here is a short code adapted from a web example that runs everywhere except for Windows/Wx4.0/Python3.6. 

```python
import wx
import wx.lib.mixins.inspection as wit

if 'phoenix' in wx.PlatformInfo:
    import wx.lib.agw.aui as aui
else:
    import wx.aui as aui

import matplotlib as mpl
from matplotlib.backends.backend_wxagg import FigureCanvasWxAgg as Canvas
from matplotlib.backends.backend_wxagg import NavigationToolbar2Wx as Toolbar


class Plot(wx.Panel):
    def __init__(self, parent, id=-1, dpi=None, **kwargs):
        wx.Panel.__init__(self, parent, id=id, **kwargs)
        self.figure = mpl.figure.Figure(dpi=dpi, figsize=(2, 2))
        self.canvas = Canvas(self, -1, self.figure)
        self.toolbar = Toolbar(self.canvas)
        self.toolbar.Realize()

        sizer = wx.BoxSizer(wx.VERTICAL)
        sizer.Add(self.canvas, 1, wx.EXPAND)
        sizer.Add(self.toolbar, 0, wx.LEFT | wx.EXPAND)
        self.SetSizer(sizer)


class PlotNotebook(wx.Panel):
    def __init__(self, parent, id=-1):
        wx.Panel.__init__(self, parent, id=id)
        self.nb = aui.AuiNotebook(self)
        sizer = wx.BoxSizer()
        sizer.Add(self.nb, 1, wx.EXPAND)
        self.SetSizer(sizer)

    def add(self, name="plot"):
        page = Plot(self.nb)
        self.nb.AddPage(page, name)
        return page.figure


def demo():
    # alternatively you could use
    #app = wx.App()
    # InspectableApp is a great debug tool, see:
    # http://wiki.wxpython.org/Widget%20Inspection%20Tool
    app = wit.InspectableApp()
    frame = wx.Frame(None, -1, 'Plotter')
    plotter = PlotNotebook(frame)
    axes1 = plotter.add('figure 1').gca()
    axes1.plot([1, 2, 3], [2, 1, 4])
    axes2 = plotter.add('figure 2').gca()
    axes2.plot([1, 2, 3, 4, 5], [2, 1, 4, 2, 3])
    frame.Show()
    app.MainLoop()

if __name__ == "__main__":
    demo()
```

**Actual outcome**

The call to ```parent.AddTool()``` in wx_compat.py throws an exception due to having the wrong arguments. It appears that the backend assumes on Windows that auiToolBar is always in use. 

**Matplotlib version**

  * Operating system: Windows-10
  * Matplotlib version: 2.0.2
  * Matplotlib backend: WxPython (4.0.0b2)
  * Python version: 3.6

default conda install, with wxPython from pip.

Patch that fixes the problem (diff on matplotlib/backends/wx_compat.py)

```diff /Users/toby/conda3//lib/python3.6/site-packages/matplotlib/backends/wx_compat.py /tmp/
148,153c148,156
<         parent.AddTool(wx_ids[text], label=text,
<                        bitmap=bmp,
<                        bmpDisabled=wx.NullBitmap,
<                        shortHelpString=text,
<                        longHelpString=tooltip_text,
<                        kind=kind)
---
>         try:
>             parent.AddTool(wx_ids[text], label=text,
>                            bitmap=bmp,
>                            bmpDisabled=wx.NullBitmap,
>                            shortHelpString=text,
>                            longHelpString=tooltip_text,
>                            kind=kind)
>         except TypeError:   #not auiToolBar!
>             parent.AddTool(wx_ids[text], text, bmp, tooltip_text)
```

**Comment by briantoby** [comment]
Here is the traceback

```
Traceback (most recent call last):
  File "tool.py", line 58, in <module>
    demo()
  File "tool.py", line 50, in demo
    axes1 = plotter.add('figure 1').gca()
  File "tool.py", line 37, in add
    page = Plot(self.nb)
  File "tool.py", line 19, in __init__
    self.toolbar = Toolbar(self.canvas)
  File "C:\work\Anaconda3-64\lib\site-packages\matplotlib\backends\backend_wx.py", line 1582, in __init__
    NavigationToolbar2.__init__(self, canvas)
  File "C:\work\Anaconda3-64\lib\site-packages\matplotlib\backend_bases.py", line 2760, in __init__
    self._init_toolbar()
  File "C:\work\Anaconda3-64\lib\site-packages\matplotlib\backends\backend_wx.py", line 1609, in _init_toolbar
    tooltip_text)
  File "C:\work\Anaconda3-64\lib\site-packages\matplotlib\backends\wx_compat.py", line 153, in _AddTool
    kind=kind)
TypeError: ToolBar.AddTool(): arguments did not match any overloaded call:
  overload 1: argument 1 has unexpected type 'int'
  overload 2: 'bmpDisabled' is not a valid keyword argument
  overload 3: 'shortHelpString' is not a valid keyword argument

```

**Comment by anntzer** [comment]
Can you try 2.1? (https://github.com/matplotlib/matplotlib/pull/9187/)

**Comment by briantoby** [comment]
Install using

conda install -c conda-forge matplotlib

?

On Oct 11, 2017, at 1:01 PM, Antony Lee <notifications@github.com<mailto:notifications@github.com>> wrote:


Can you try 2.1? (#9187<https://github.com/matplotlib/matplotlib/pull/9187>)

**Comment by anntzer** [comment]
yes
attn @RobinD42 just in case...

**Comment by briantoby** [comment]
Yes, this was fixed in 2.10. Thanks!
